services/repository-managers/src/test/java/org/sagebionetworks/repo/manager/file/MultipartManagerV2ImplAutowireTest.java

```diff
@@ -546,7 +551,77 @@ public void testMultipartUploadCopyFromMultipartUploadWithMultiplePartsSinglePar
 		// Uses the max part size to do a one shot copy
 		doMultipartCopy(sourceEntity, copyDestination, PartUtils.MAX_PART_SIZE_BYTES);
 	}
-	
+
+	@Test
+	public void testS3StartOrResumeMultipartRequestWithObjectDeleted() throws Exception {
+		String fileName = "foo.txt";
+		String contentType = "plain/text";
+		String fileContent = "This is the content of the file";
+		Long storageLocationId = null;
+		boolean useContentTypeForParts = false;
+
+		S3FileHandle sourceFileHandle = (S3FileHandle) doMultipartUpload(fileName, contentType, fileContent, storageLocationId, useContentTypeForParts, true);
+
+		s3Client.deleteObject(sourceFileHandle.getBucketName(), sourceFileHandle.getKey());
+
+		assertFalse(s3Client.doesObjectExist(sourceFileHandle.getBucketName(), sourceFileHandle.getKey()));
+
+		// Call under test
+		S3FileHandle resultFileHandle = (S3FileHandle) doMultipartUpload(fileName, contentType, fileContent, storageLocationId, useContentTypeForParts, false);
+
+		assertNotEquals(sourceFileHandle, resultFileHandle);
+	}
+
+	@Test
+	public void testGCStartOrResumeMultipartRequestWithObjectDeleted() throws Exception {
+		Assume.assumeTrue(stackConfiguration.getGoogleCloudEnabled());
+
+		String fileName = "foo.txt";
+		String contentType = "plain/text";
+		String fileContent = "This is the content of the file";
+		Long storageLocationId = null;
+		boolean useContentTypeForParts = false;
+
+		GoogleCloudFileHandle sourceFileHandle = (GoogleCloudFileHandle) doMultipartUpload(fileName, contentType, fileContent, storageLocationId, useContentTypeForParts, true);
+
+		googleCloudStorageClient.deleteObject(sourceFileHandle.getBucketName(), sourceFileHandle.getKey());
+
+		assertFalse(s3Client.doesObjectExist(sourceFileHandle.getBucketName(), sourceFileHandle.getKey()));
+
+		// Call under test
+		GoogleCloudFileHandle resultFileHandle = (GoogleCloudFileHandle) doMultipartUpload(fileName, contentType, fileContent, storageLocationId, useContentTypeForParts, false);

```

Comment:
Please take the forceRestart value out in its own variable so we know what the value when reading the code